client-base/src/main/webapp/html/javascript/data-utils.js

```diff
@@ -612,14 +613,93 @@ function matchCounter() {
 				  $('#suggestor_search').css('display', 'none');
 				}
             }).fail(function (xhr,status,errorlog) {
-                //$('#matchesResultDiv').html('<span class="redText">The KnetMiner server is currently offline. Please try again later.</span>');
-                var server_error= JSON.parse(xhr.responseText); // full error json from server
-                var errorMsg= server_error.statusReasonPhrase +" ("+ server_error.title +")";
-                $('#matchesResultDiv').html('<span class="redText">'+errorMsg+'</span>');
-                console.log(server_error.detail); // detailed stacktrace
+                ErrorComponent('#matchesResultDiv',xhr)
             });
         } else {
             $('#matchesResultDiv').html('');
         }
     }
 }
+
+
+
+/*
+ * Function to create,get and showcase gene name synonym with a dropdown onclick event
+ *
+ */
+function createGeneNameSynonyms(ondexIds){
+
+    var geneNameSynonyms; 
+    var status; 
+    var synonymNameUrl = `http://localhost:9090/ws/graphinfo/concept-info?ids=${ondexIds}`;
```

Comment:
Hard coded to localhost:9090 ?

@manlikleo @marco-brandizi

Reply:
@manlikleo, yes, that is wrong, `http://<host>:<port>/ws/<dataset-name>` is dynamic, it varies depending on the configuration and can't be hardcoded, but must be taken from `api_url`, which is defined in [utils-config.js][10] (its final value is instantiated by Maven via [Maven filtering][20]).

[10]: https://github.com/Rothamsted/knetminer/blob/master/aratiny/aratiny-client/src/main/webapp/html/javascript/utils-config.js
[20]: https://maven.apache.org/plugins/maven-resources-plugin/examples/filter.html

Reply:
I tried api_url without knowing base_api_url existed. It's fixed. I'm sorry for making such an unethical mistake